libs/server/ExpireOption.cs

```diff
@@ -13,23 +13,31 @@ public enum ExpireOption : byte
         /// <summary>
         /// None
         /// </summary>
-        None,
+        None = 0,
         /// <summary>
         /// Set expiry only when the key has no expiry
         /// </summary>
-        NX,
+        NX = 1 << 0,
         /// <summary>
         /// Set expiry only when the key has an existing expiry 
         /// </summary>
-        XX,
+        XX = 1 << 1,
         /// <summary>
         /// Set expiry only when the new expiry is greater than current one
         /// </summary>
-        GT,
+        GT = 1 << 2,
         /// <summary>
         /// Set expiry only when the new expiry is less than current one
         /// </summary>
-        LT
+        LT = 1 << 3,
+        /// <summary>
+        /// Set expiry only when the key has an existing expiry and the new expiry is greater than current one
+        /// </summary>
+        XXGT = XX | GT,
```

Comment:
NX and GT / LT are mutually exclusive, so the presence of GT / LT implies XX

Reply:
It's not the case. Below is an example. when values are not there LT and GT will treat it as infinite expiration time but with `XX` setting it doesn't, it will not apply the expiration time if there is no TTL for the key currently.
![image](https://github.com/user-attachments/assets/b9271cf9-e8ab-4666-952c-f88923a98c80)

Reply:
> It's not the case. Below is an example. when values are not there LT and GT will treat it as infinite expiration time but with `XX` setting it doesn't, it will not apply the expiration time if there is no TTL for the key currently. ![image](https://private-user-images.githubusercontent.com/24755596/369548885-b9271cf9-e8ab-4666-952c-f88923a98c80.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY4NzQ3MjQsIm5iZiI6MTcyNjg3NDQyNCwicGF0aCI6Ii8yNDc1NTU5Ni8zNjk1NDg4ODUtYjkyNzFjZjktZThhYi00NjY2LTk1MmMtZjg4OTIzYTk4YzgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIwVDIzMjAyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYzZjBjMTRhNGFjODdlNDAyYzRmNjZlNjk0MDlkNzllNWQ5NWU5Mzc1OTkzYjJjYmZjOWI3OWI3OTQ4ZjQ5Y2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Rb3lIO-ZLsHig1hqC-dTyEH2AwKMOoy1cEilBGgq7aA)

Oh that's good to know, so I guess only GT implies XX then?

Reply:
> so I guess only GT implies XX then?

Technically no but practically yes because nothing can be bigger than infinite.

Can I mark this review comments as resolved without making any change?